### ISSUE TITLE
Ansible-playbook run seems to use old data

### ISSUE THREAD

**Issue by rndmh3ro** [ansible/awx#1469]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - API

##### SUMMARY
Ansible run seems to use old data

##### ENVIRONMENT
* AWX version: 1.0.4.14
* AWX install method: docker on linux
* Ansible version: 2.4.3.0
* Operating System: rhel7
* Web Browser: firefox

##### STEPS TO REPRODUCE

I have a playbook that creates users on the target-servers. There's user `foo`. This user had id `500` sometime in the past. Then I changed the id on the target-servers to `505` and also in the git-repository (see below). Now everytime the playbook runs via awx, the user-id wants to be changed to `500`.

I verified in the docker-container that the project uses the correct git checkout. I even deleted the project and recreated it. awx also tells me that its using the correct commit id, where the user-id is `505`.

Playbook:
```
- hosts: all
  roles:
   - role: user
```

user-role:
```
# file: user/tasks/main.yml

- name: generic-users | Make sure the users are present
  user:
    name: "{{ item.name }}"
    group: "{{ item.group|default(omit) }}"
    groups: "{{ ','.join(item.groups)|default(omit) }}"
    append: "{{ item.append|default(omit) }}"
    password: "{{ item.pass|default(omit) }}"
    comment: "{{ item.comment|default(omit) }}"
    shell: "{{ item.shell|default(omit) }}"
    uid: "{{ item.uid|default(omit) }}"
    home: "{{ item.home|default(omit) }}"
    system: "{{ item.system|default(omit) }}"
    state: present
  with_items: "{{ genericusers_users }}"
```

group_vars:
```
genericusers_users:
  - name: "foo"
    group: "foo"
    groups: []
    shell: "/bin/bash"
    home: "/home/foo"
    append: "no"
    comment: "foo User"
    uid: "505"
    system: "yes"
    ssh_keys: []
```

If I run this playbook on the shell from within the docker container, it works flawlessly, meaning the user-id does not get changed to 500.
But still when running it from awx, the user-id gets changed.

I restarted all the docker-container and the server running docker, too. It did not help.

What did help is to set the `genericusers_users:` in awx as an extra variable. Then the id stays at `505`.

I guess there's some cache here at work?

##### EXPECTED RESULTS

The user-id does not get changed.

##### ACTUAL RESULTS

see above.

##### ADDITIONAL INFORMATION

![grafik](https://user-images.githubusercontent.com/3198961/37150752-2e0be182-22d3-11e8-9f4b-04e71563db22.png)

![grafik](https://user-images.githubusercontent.com/3198961/37150767-3fb0fd32-22d3-11e8-8874-36bc1be9a8a2.png)

![grafik](https://user-images.githubusercontent.com/3198961/37150787-5fc2fa1c-22d3-11e8-9d01-290b114896f3.png)

```
{
    "_ansible_parsed": true,
    "changed": false,
    "_ansible_no_log": false,
    "_ansible_item_result": true,
    "invocation": {
        "module_args": {
            "comment": "foo User",
            "ssh_key_bits": 0,
            "update_password": "always",
            "non_unique": false,
            "force": false,
            "uid": "500",
            "ssh_key_passphrase": null,
            "createhome": true,
            "home": "/home/foo",
            "skeleton": null,
            "append": false,
            "ssh_key_type": "rsa",
            "ssh_key_comment": "ansible-generated on server",
            "group": "foo",
            "system": false,
            "state": "present",
            "local": null,
            "shell": "/bin/bash",
            "expires": null,
            "ssh_key_file": null,
            "groups": [
                ""
            ],
            "move_home": false,
            "password": null,
            "name": "foo",
            "seuser": null,
            "remove": false,
            "login_class": null,
            "generate_ssh_key": null
        }
    },
    "item": {
        "comment": "foo User",
        "shell": "/bin/bash",
        "group": "foo",
        "ssh_keys": [],
        "system": "no",
        "name": "foo",
        "groups": [],
        "home": "/home/foo",
        "append": "no",
        "uid": "500"
    },
    "rc": 8,
    "msg": "usermod: user foo is currently used by process 1937\n",
    "_ansible_ignore_errors": null,
    "name": "foo"
}
```

**Comment by shanemcd** [comment]
Did you check "Update Revision on Launch" on the Project?

**Comment by matburt** [comment]
How do you have your inventory set up?  Is it coming from the project via scm inventory? What options have you selected on the inventory source for how we treat variables during the inventory sync?

**Comment by rndmh3ro** [comment]
> How do you have your inventory set up? Is it coming from the project via scm inventory? 

Yes, via git scm.

> What options have you selected on the inventory source for how we treat variables during the inventory sync?

This was the problem. There was no inventory source anymore. I guess when I deleted and recreated the project, the inventory source disappeared, too.
However before deleting the project the problem was also there. So I guess I did not have "Update on Launch" checked before.
Now with "Update on Launch" checked, it works.

I'm not sure about the behaviour that when deleting the project the inventory kind of remains. Do I missunderstand something here?
